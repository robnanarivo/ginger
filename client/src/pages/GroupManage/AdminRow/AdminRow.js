import React from 'react';
import { useParams } from 'react-router-dom';
import Box from '@mui/material/Box';
import Avatar from '@mui/material/Avatar';
// import FolderIcon from '@mui/icons-material/Folder';
import Button from '@mui/material/Button';
import { Typography } from '@mui/material';
import { updateGroup } from '../../../api';

export default function AdminRow(props) {
  const {
    adminId,
    adminPic,
    adminName,
    creatorId,
    updateAdmins,
  } = props;
  const { groupId } = useParams();
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        p: 1,
        height: '60px',
      }}
    >
      <Avatar sx={{ marginLeft: 3 }}>
        {adminPic === ''
          ? (<Avatar alt="Admin Pic" src="" />)
          : (<Avatar alt="Admin Pic" src={adminPic} sx={{ backgroundColor: '#FFFFFF' }} />)}
      </Avatar>
      <Typography sx={{ marginLeft: 5 }}>
        {adminName}
      </Typography>
      <Box sx={{ flexGrow: 1 }} />
      {
        creatorId !== adminId
          ? (
            <Button
              variant="outlined"
              sx={{ marginLeft: 5, minWidth: '100px', marginRight: 5 }}
              onClick={async () => {
                await updateGroup(groupId, 'leave', [], [adminId]);
                updateAdmins();
              }}
            >
              REMOVE
            </Button>
          )
          : (
            <div />
          )
      }

    </Box>
  );
}
