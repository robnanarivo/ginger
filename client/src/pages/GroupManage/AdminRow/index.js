import AdminRow from './AdminRow';

export default AdminRow;
