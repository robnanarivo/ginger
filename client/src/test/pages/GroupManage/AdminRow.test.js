import React from 'react';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
// import { render, screen } from '@testing-library/react';
import AdminRow from '../../../pages/GroupManage/AdminRow/AdminRow';
import 'jest-styled-components';

describe('AdminRow unit test', () => {
  it('AdminRow Snapshot test', () => {
    const history = createMemoryHistory();
    const tree = renderer
      .create(
        <Router history={history}>
          <AdminRow />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
