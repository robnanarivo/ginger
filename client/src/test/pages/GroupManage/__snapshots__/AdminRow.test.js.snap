// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminRow unit test AdminRow Snapshot test 1`] = `
<div
  className="MuiBox-root css-3cku3k"
>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-g1xdqz-MuiAvatar-root"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-9cpfib-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1pjrok-MuiTypography-root"
  />
  <div
    className="MuiBox-root css-i9gxme"
  />
  <div />
</div>
`;
